File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product
def solution(word):
    answer = 0
    alphabet = ['A', 'E', 'I', 'O', 'U']
    dic = []
    for i in range(1,6):
        for j in product(alphabet,repeat=i):
            dic.append("".join(j))
    dic.sort()
    for i in range(len(dic)):
        if dic[i] == word:
            answer = i+1
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    var answer = 0;
    const alphabet = ['A', 'E', 'I', 'O', 'U'];

    function dfs(now) {
        if (now.length > 0) answer++;

        if (now === word) {
            return true;
        }

        if (now.length >= 5) return false;

        for (let i = 0; i < alphabet.length; i++) {
            if (dfs(now + alphabet[i])) return true;
        }

        return false;
    }

    dfs("");
    return answer;
}
